components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/OAuthClientAuthnService.java

```diff
@@ -258,4 +295,245 @@ private boolean canAuthenticate(OAuthClientAuthenticator oAuthClientAuthenticato
 
         return oAuthClientAuthenticator.canAuthenticate(request, bodyContentMap, oAuthClientAuthnContext);
     }
+
+    /**
+     * Validate whether the client authentication method of the request is registered for the application.
+     *
+     * @param request                Http servlet request.
+     * @param contentParams          Map of request body params.
+     * @return Whether the client authentication method of the request is registered for the application.
+     * @throws OAuthClientAuthnException
+     */
+    private boolean isRegisteredClientAuthMethod(HttpServletRequest request, Map<String, List> contentParams)
+            throws OAuthClientAuthnException {
+
+        String registeredClientAuthMethod = retrieveRegisteredAuthMethod(extractClientId(request, contentParams));
+
+        if (registeredClientAuthMethod.equals(OAuthConstants.NOT_APPLICABLE)) {
```

Comment:
swap the constant with the variable

Reply:
Fixed in 36cde69f01a0865dc525ff8c963f66e0af53cf89